fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/size/CitusSizeCollector.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+ 
+package com.ibm.fhir.schema.size;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.schema.app.util.SchemaSupport;
+
+/**
+ * Collect size metrics for a Citus database and add them
+ * to the model
+ * @implNote see https://github.com/citusdata/citus_docs/issues/231
+ */
+public class CitusSizeCollector implements ISizeCollector {
+    private static final Logger logger = Logger.getLogger(PostgresSizeCollector.class.getName());
+
+    // the model used to collect the size info
+    private final FHIRDbSizeModel model;
+
+    /**
+     * Public constructor
+     * @param model
+     */
+    public CitusSizeCollector(FHIRDbSizeModel model) {
+        this.model = model;
+    }
+
+    @Override
+    public void run(String schemaName, Connection connection, IDatabaseTranslator translator) {
+        collectTableInfo(schemaName, connection, translator);
+        collectIndexInfo(schemaName, connection, translator);
+    }
+
+    private void collectTableInfo(String schemaName, Connection connection, IDatabaseTranslator translator) {
+        final String select = ""
+                + "SELECT  table_name, "
+                + "        0 AS row_estimate, "
+                + "        total_bytes, "
+                + "        total_bytes - table_bytes AS index_bytes, "
+                + "        0 AS toast_bytes, "
+                + "        table_bytes "
+                + "   FROM (SELECT c.oid, "
+                + "                nspname AS table_schema, "
+                + "                relname AS table_name, "
+                + "                citus_total_relation_size(p.logicalrelid) AS total_bytes, "
+                + "                citus_table_size(p.logicalrelid) AS table_bytes "
+                + "           FROM pg_class c "
+                + "      LEFT JOIN pg_namespace n "
+                + "             ON n.oid = c.relnamespace "
+                + "      LEFT JOIN pg_dist_partition p "
+                + "             ON p.logicalrelid = (nspname || '.' || relname)::regclass "
+                + "          WHERE relkind = 'r' "
+                + "            AND nspname = ?"
+                + "   ) a";
```

Comment:
is the trailing `a` intended?

Reply:
Yes. It's the alias for the sub-query. Required in some cases.